deeplake/core/chunk_engine.py

```diff
@@ -1029,7 +1038,14 @@ def extend(
             if link_callback:
                 if not isinstance(verified_samples, np.ndarray):
                     samples = [
-                        None if is_empty_list(s) else s for s in verified_samples
+                        None
+                        if is_empty_list(s)
+                        or (
+                            isinstance(s, deeplake.core.tensor.Tensor)
+                            and s.is_empty_tensor
+                        )
+                        else s
+                        for s in verified_samples
```

Comment:
To be honest, it would be much easier to read if you remove list comprehension and use regular for loop instead